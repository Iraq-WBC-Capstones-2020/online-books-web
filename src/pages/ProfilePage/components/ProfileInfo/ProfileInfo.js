import React from 'react';
import { MDBContainer, MDBRow, MDBCol } from 'mdbreact';
import { useTranslation } from 'react-i18next';

function ProfileInfo({ profileInfo }) {
    const { t } = useTranslation();

  return (
    <MDBContainer>
      <MDBRow className="h-auto">
        <MDBCol lg="3" md="6" sm="12" size="12" className="py-4"></MDBCol>
        <MDBCol
          lg="8"
          md="6"
          sm="12"
          size="12"
          className="py-4 text-center text-md-left"
        >
          <h4 className="text-3xl font-medium text-blue-500 pb-3">
            {profileInfo.name}
          </h4>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  );
}

export default ProfileInfo;
