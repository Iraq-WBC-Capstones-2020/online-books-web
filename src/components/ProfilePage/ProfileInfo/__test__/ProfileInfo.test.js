import React from 'react';
import renderer from 'react-test-renderer';
import ProfileInfo from '../ProfileInfo';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
test('ProfileInfo snapshot', () => {
  jest.mock('react-i18next', () => ({
    useTranslation: () => ({ t: (key) => key }),
  }));
  const history = createMemoryHistory();
  const profileInfo = {
    id: '120147514545',
    name: 'Sara Ahmed',
    email: 'sara@eample.com',
    img: 'https://via.placeholder.com/300x300',
    gender: 'Female',
    joined: 886280400000,
    birthday: 886291200000,
    phoneNumber: '07154245455',
    address: {
      street: 'Middle Erbil',
      city: 'Erbil',
      country: 'Iraq',
    },
  };
  const tree = renderer
    .create(
      <Router history={history}>
        <ProfileInfo profileInfo={profileInfo} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
