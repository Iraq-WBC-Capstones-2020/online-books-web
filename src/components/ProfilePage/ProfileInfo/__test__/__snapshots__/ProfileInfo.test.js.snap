// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileInfo snapshot 1`] = `
Array [
  <button
    className="btn-transparent btn Ripple-parent shadow-none text-blue-500 p-0 m-0 float-right mr-2 capitalize"
    data-test="button"
    onClick={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    profilePage.edit
    <div
      className="Ripple "
      data-test="waves"
      style={
        Object {
          "height": "0px",
          "left": "0px",
          "top": "0px",
          "width": "0px",
        }
      }
    />
  </button>,
  <div
    className="clearfix"
  />,
  <div
    className="row"
    data-test="row"
  >
    <div
      className="col-12 col-sm-12 col-md-6"
      data-test="col"
    >
      <h3
        className="text-2xl leading-loose font-semibold"
      >
        profilePage.basicInfo
      </h3>
      <div
        className="row"
        data-test="row"
      >
        <div
          className="col-6 col-sm-6 col-md-4"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            profilePage.name
            :
          </h3>
        </div>
        <div
          className="col-6 col-sm-6 col-md-8"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            Sara Ahmed
          </h3>
        </div>
      </div>
      <div
        className="row"
        data-test="row"
      >
        <div
          className="col-6 col-sm-6 col-md-4"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            profilePage.gender
            :
          </h3>
        </div>
        <div
          className="col-6 col-sm-6 col-md-8"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            Female
          </h3>
        </div>
      </div>
      <div
        className="row"
        data-test="row"
      >
        <div
          className="col-6 col-sm-6 col-md-4"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            profilePage.birthday
            :
          </h3>
        </div>
        <div
          className="col-6 col-sm-6 col-md-8"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            Sun Feb 01 1998
          </h3>
        </div>
      </div>
      <div
        className="row"
        data-test="row"
      >
        <div
          className="col-6 col-sm-6 col-md-4"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            profilePage.city
            :
          </h3>
        </div>
        <div
          className="col-6 col-sm-6 col-md-8"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            Erbil
          </h3>
        </div>
      </div>
      <div
        className="row"
        data-test="row"
      >
        <div
          className="col-6 col-sm-6 col-md-4"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            profilePage.country
            :
          </h3>
        </div>
        <div
          className="col-6 col-sm-6 col-md-8"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            Iraq
          </h3>
        </div>
      </div>
    </div>
    <div
      className="col-12 col-sm-12 col-md-6"
      data-test="col"
    >
      <h3
        className="text-2xl leading-loose font-semibold"
      >
        profilePage.contactInfo
      </h3>
      <div
        className="row"
        data-test="row"
      >
        <div
          className="col-6 col-sm-6 col-md-4"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            profilePage.address
            :
          </h3>
        </div>
        <div
          className="col-6 col-sm-6 col-md-8"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            Middle Erbil
          </h3>
        </div>
      </div>
      <div
        className="row"
        data-test="row"
      >
        <div
          className="col-6 col-sm-6 col-md-4"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            profilePage.phoneNumber
            :
          </h3>
        </div>
        <div
          className="col-6 col-sm-6 col-md-8"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            07154245455
          </h3>
        </div>
      </div>
      <div
        className="row"
        data-test="row"
      >
        <div
          className="col-6 col-sm-6 col-md-4"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            profilePage.email
            :
          </h3>
        </div>
        <div
          className="col-6 col-sm-6 col-md-8"
          data-test="col"
        >
          <h3
            className="text-base leading-loose"
          >
            sara@eample.com
          </h3>
        </div>
      </div>
    </div>
  </div>,
]
`;
